tests/test_templates_layers.py

```diff
@@ -0,0 +1,684 @@
+                         [ 0.36293094, 1.30725604, 0.11578591, 1.5983082 ]]),
+                np.array([[ 6.21267547, 3.71076099, 0.34060195, 2.86031556],
+                         [ 3.20443756, 6.26536946, 6.18450567, 1.50406923]]),
+                np.array([[ 0.1376345 , 0.22541113, 0.14306356, 0.13019402],
+                         [ 0.26999146, 0.26256351, 0.14722687, 0.23137066]])
+            ]
+
+    def test_cvneuralnet_integration(self, num_subsystems, weights, depth, N):
+        """integration test for the CVNeuralNetLayers method."""
+        dev = DummyDevice(wires=num_subsystems)
+
+        def circuit(weights):
+            CVNeuralNetLayers(*weights, wires=range(N))
+            return qml.expval.X(wires=0)
+
+        qnode = qml.QNode(circuit, dev)
+
+        # execution test
+        qnode(weights)
+        queue = qnode.queue
+
+        # Test that gates appear in the right order for each layer:
+        # BS-R-S-BS-R-D-K
+        for l in range(depth):
+            gates = [qml.Beamsplitter, qml.Rotation, qml.Squeezing,
+                     qml.Beamsplitter, qml.Rotation, qml.Displacement]
+
+            # count the position of each group of gates in the layer
+            num_gates_per_type = [0, 6, 4, 4, 6, 4, 4, 4]
+            s = np.cumsum(num_gates_per_type)
+            gc = l*sum(num_gates_per_type)+np.array(list(zip(s[:-1], s[1:])))
+
+            # loop through expected gates
+            for idx, g in enumerate(gates):
+                # loop through where these gates should be in the queue
+                for opidx, op in enumerate(queue[gc[idx, 0]:gc[idx, 1]]):
+                    # check that op in queue is correct gate
+                    assert isinstance(op, g)
+
+                    # test that the parameters are correct
+                    res_params = op.parameters
+
+                    if idx == 0:
+                        # first BS
+                        exp_params = [weights[0][l][opidx], weights[1][l][opidx]]
+                    elif idx == 1:
+                        # first rot
+                        exp_params = [weights[2][l][opidx]]
+                    elif idx == 2:
+                        # squeezing
+                        exp_params = [weights[3][l][opidx], weights[4][l][opidx]]
+                    elif idx == 3:
+                        # second BS
+                        exp_params = [weights[5][l][opidx], weights[6][l][opidx]]
+                    elif idx == 4:
+                        # second rot
+                        exp_params = [weights[7][l][opidx]]
+                    elif idx == 5:
+                        # displacement
+                        exp_params = [weights[8][l][opidx], weights[9][l][opidx]]
+
+                    assert res_params == exp_params
+
+
+class TestStronglyEntangling:
+    """Tests for the StronglyEntanglingLayers method from the pennylane.templates.layers module."""
+
+    def test_integration(self):
```

Comment:
Same as above, always better to parametrize rather than use for loops over assert statements:

```python
@pytest.mark.parametrize('num_wires', range(2, 4))
def test_integration(self, num_wires):
```

Reply:
I did only minimum pytest conversion for Christian's code...Implemented your suggestion, but there is more to do. Guess not a priority.